src/custom/utils/trade.ts

```diff
@@ -10,6 +10,7 @@ import { EcdsaSignature, Signature } from '@gnosis.pm/gp-v2-contracts'
 
 const DEFAULT_SIGNING_SCHEME = SigningScheme.EIP712
 const METAMASK_SIGNATURE_ERROR_CODE = -32603
+const V4_ERROR_MSG_REGEX = /eth_signTypedData_v4 does not exist/i
```

Comment:
Do we need a regex for this? it feels like equality on the message would do, right?

Reply:
okoko,
u know i love regex, all good. Just for trolling a bit https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith